### PR TITLE
Feat/refactor sprite groups

### PR DESCRIPTION
**Pull request by alecpramanik**
# Proposing the following few major changes for better encapsulation:
## TLDR: Separate sprite group initialization and handling per file instead of doing it in `main.py`. `main.py` would only need to initialize the relevant classes and call their `draw` methods and `set_value`/`set_background` methods without worrying about passing sprite groups to these methods.

- Move handling of sprite groups for classes in `grid.py`, `menu.py` and `settings.py` in their own files out of `main.py`.
  * Make `Grid` initialize and  store its own sprite group dictionary which is passed to other classes in `grid.py` and used to draw the `Grid` onto a pygame surface in a new `draw_grid` method.
  * This change will make class methods such as `set_value` and `set_background` only need the relevant value as an argument without also having to pass the sprite groups to them.
  * Make `Square` store references to the relevant sprite groups for `SquareBackground` and `Number`. These references are passed to these classes as single groups. The reasoning behind this is that these two classes are inherently tied to `Square` and shouldn't have any need for any other sprite groups in the dictionary. The full dict should still be passed to `Box` and `Square` since they make use of more than one group (because of `Square's` passing to the background and number classes).
  * As mentioned above, a new `draw_grid` method can be added to `Grid` for use outside of the `grid.py` by `main.py`
- Similar changes can be done to `menu.py` and `settings.py`.
  * Initialize and store the relevant sprite groups in `Menu` and `Settings`.
  * Make no further changes to the constructors of other classes in these files except changing the `groups` argument to `sprite_groups` for clarity.
  * Add `draw_menu` and `draw_settings` methods to these classes.
- Move `draw` function out of `main.py` and into `helper_utils.py`. This can then be used by `Grid`, `Menu` and `Settings` in their draw methods.
- Update `main.py` to take these changes into consideration.

# The above changes can be merged with this PR.

### CODE REVIEW

**Comment by BevapnadeBerit** [comment]
I support this change.

The questions I have regard:
- Did you mean to call menu.draw_settings instead of settings.draw_settings in main.py line 161? If that was a mistake, changes should be made at menu.py line 46.
- Connected to that, is it necessary to store menu sprites in settings.py and settings (line 36) sprites in menu.py (line 17)?

**Comment by alecpramanik** [comment]
> I support this change.
> 
> The questions I have regard:
> 
>     * Did you mean to call menu.draw_settings instead of settings.draw_settings in main.py line 161? If that was a mistake, changes should be made at menu.py line 46.
> 
>     * Connected to that, is it necessary to store menu sprites in settings.py and settings (line 36) sprites in menu.py (line 17)?

Indeed, it should be `settings.draw_settings` in `main.py`.

You're right. Since I wasn't super familiar with `menu.py` and `settings.py`, I decided to make both sprite groups accessible in each file, but I see now that this may be counterproductive since they are being initialized in each class. Each sprite group should only initialized once, in their respective classes. However, I think we should keep these sprite groups in dicts despite these files only requiring one group, in case we need more.

So, like this?
```python
        self.sprite_groups = {
            "menu": pygame.sprite.Group(),
        }
```
```python
        self.sprite_groups = {
            "settings": pygame.sprite.Group(),
        }
```

**Comment by BevapnadeBerit** [comment]
Good!
